src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int linkedListSize;
+
+    public MyLinkedList() {
+        linkedListSize = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (linkedListSize == 0) {
+            Node<T> headNode = new Node<>(null, value, null);
+            head = headNode;
+            tail = headNode;
+            linkedListSize++;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+            linkedListSize++;
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == linkedListSize) {
+            add(value);
+            return;
+        }
+        Node<T> findNode = head;
+        for (int i = 0; i < linkedListSize; i++) {
+            if (i == index) {
+                Node<T> prevNode = findNode.prev;
+                Node<T> newNode = new Node<>(prevNode, value, findNode);
+                if (prevNode != null) {
+                    prevNode.next = newNode;
+                } else {
+                    head = newNode;
+                }
+                findNode.prev = newNode;
+                linkedListSize++;
+                return;
+            }
+            findNode = findNode.next;
+        }
+        throw new IndexOutOfBoundsException("Index [" + index + "] does not exist!");
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> findNode = head;
+        for (int i = 0; i < linkedListSize; i++) {
+            if (i == index) {
+                return findNode.item;
+            }
+            findNode = findNode.next;
+        }
+        throw new IndexOutOfBoundsException("Index [" + index + "] does not exist!");
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> findNode = head;
+        for (int i = 0; i < linkedListSize; i++) {
+            if (i == index) {
+                T oldValue = findNode.item;
+                findNode.item = value;
+                return oldValue;
+            }
+            findNode = findNode.next;
+        }
+        throw new IndexOutOfBoundsException("Index [" + index + "] does not exist!");
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> findNode = head;
+        for (int i = 0; i < linkedListSize; i++) {
+            if (i == index) {
+                delete(findNode);
+                return findNode.item;
+            }
+            findNode = findNode.next;
```

Comment:
I would remove finding the node by index to separate method since you are using it in remove(), get() set() methods